_posts/2017-09-22-redis-2-to-redis-3.md

```diff
@@ -0,0 +1,48 @@
+---
+layout:     post
+title:      Our journey from Redis 2 to Redis 3 while not taking the site down.
+date:       2017-09-22
+summary:    How we managed to upgrade from Redis 2 to Redis 3 without causing a site outage.
+category:   Devops
+tags:       redis
+author:     craig_stewart
+---
+We use Redis within Sky Betting & Gaming as a shared in-memory cache for things like identity tokens that need to be known across api servers, or web servers. Within the Core Tribe this is used to help deal with the huge number of logins we have to handle day to day and particularly at busy times when we could have more than 20,000 people logging in within a single minute. This works well in so far as the data is readily available to a large number of servers (in the case of SSO tokens 70 Apache HTTPD servers). We've recently embarked upon a process of upgrading our redis servers, and this upgrade is intended to enable the use of the native clustering features available from redis 3.2. This blog post hopes to explain why we're using clustering, what problems we have encountered along the way, and what our solutions have been.
```

Comment:
"Redis" and "redis" are both used throughout the article, I suggest standardising on "Redis"

Reply:
Similar to the first mention of twemproxy should I link to the redis website on the first use of Redis? (and possibly other techs like Node.js and PHP?)

Reply:
I'd say yes for Redis, but personally I think Node and PHP are probably well-known enough to not need a link.